framework/PostProcessors/FTstructure.py

```diff
@@ -0,0 +1,160 @@
+
+    if not self.topEventID in self.gateID:
+      raise IOError('FTImporterPostProcessor: specified top event ' + str(self.topEventID) + ' is not contained in the fault-tree; available gates are: ' + str(self.gateID))
+
+  def returnDict(self):
+    """
+      This method returns
+      @ In,  None
+      @ Out, outcome, dict, dictionary containing
+    """
+    self.FTsolver()
+    outcome = self.constructData()
+    return outcome
+
+  def FTsolver(self):
+    """
+      This method determines the ordered sequence of gates to compute in order to solve the full FT.
+      The determined ordered sequence is stored in self.gateSequence.
+      @ In,  None
+      @ Out, None
+    """
+    self.gateSequence = []
+    availBasicEvents = copy.deepcopy(self.basicEvents)
+    availBasicEvents = availBasicEvents + self.houseEvents.keys()
+    counter = 0
+    while True:
+      complete=False
+      for gate in self.gateList.keys():
+        if set(self.gateList[gate].returnArguments()) <= set(availBasicEvents):
+          self.gateSequence.append(gate)
+          availBasicEvents.append(gate)
+        if set(availBasicEvents) == set(self.basicEvents+self.gateID+self.houseEvents.keys()):
+          complete=True
+          break
+        if counter > len(self.gateList.keys()):
+          raise IOError('FTImporterPostProcessor Post-Processor ' + self.name + ': the provided FT cannot be computed')
+        counter += 1
+      if complete:
+        break
+
+  def evaluateFT(self,combination):
+    """
+      This method determines the outcome of the FT given a set of basic-event values
+      @ In,  combination, dict, dictionary containing values for all basic-events
+      @ Out, values, dict, dictionary containing calculated values for all gates
+    """
+    values = {}
+    for gate in self.gateSequence:
+      values[gate] = self.gateList[gate].evaluate(combination)
+      combination[gate] = values[gate]
+    return values
+
+  def constructData(self):
+    """
+      This method determines the outcome of the FT given a set of basic-event values
+      @ In,  None
+      @ Out, outcome, dict, dictionary containing calculated values for all basic-events and the Top-event
+    """
+    combinations = list(itertools.product([0,1],repeat=len(self.basicEvents)))
+    outcome={}
+    outcome={key:np.zeros(len(combinations)) for key in self.basicEvents}
+    outcome[self.topEventID] = np.zeros(len(combinations))
+    for index,combination in enumerate(combinations):
+      combinationDict = {key: combination[index] for index,key in enumerate(self.basicEvents)}
+      for houseEvent in self.houseEvents.keys():
+        combinationDict[houseEvent] = self.houseEvents[houseEvent]
+      out = self.evaluateFT(combinationDict)
+      for key in self.basicEvents:
+        outcome[key][index]=float(combinationDict[key])
+      outcome[self.topEventID][index] = out[self.topEventID]
+    return outcome
+
+def findAllRecursive(node, element):
```

Comment:
it is redundant.... you define the same method in two different modules... you should have it in the utils module and actually be sure that you need it